From d79006dc5f7ed39616badc85ecfaf72043e78753 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 22 Dec 2024 19:14:53 +0200
Subject: [PATCH] [SW-214515] gaudi3: pcie: diasble tx_precoding

Disable PCIe tx_precoding in the PCI configuration space for better link
stability. This change is requested by the SiVal team.

Change-Id: I19b9eb0914a5f76c062eed52fe80b703aacb36f7
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 38eb258..2f0c9e9 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1039,6 +1039,10 @@
 			PCIE_DBI_PL32G_CONTROL_REG_EQ_BYPASS_HIGHEST_RATE_DISABLE_M);
 	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PL32G_CAPABILITY_REG), 0,
 			PCIE_DBI_PL32G_CAPABILITY_REG_EQ_BYPASS_HIGHEST_RATE_SUPPORT_M);
+
+	/* disable tx precoding */
+	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PL32G_STATUS_REG), 0,
+			PCIE_DBI_PL32G_STATUS_REG_TX_PRECODE_REQ_M);
 }
 
 __weak void preboot_update_bar_sizes(void) {}
